src/utils/indexer.hpp

```diff
@@ -97,18 +103,27 @@ struct Indexer {
 
   KOKKOS_FORCEINLINE_FUNCTION
   auto GetIdxArray(int idx) const {
-    return get_array_from_tuple(
-        GetIndicesImpl(idx, std::make_index_sequence<sizeof...(Ts)>()));
+    return GetIndicesArrayImpl(idx, std::make_index_sequence<sizeof...(Ts)>());
+  }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  auto GetIdxKArray(int idx) const {
+    return GetIndicesKArrayImpl(idx, std::make_index_sequence<sizeof...(Ts)>());
+  }
```

Comment:
I would just call this `GetIdxArray` and have it always return a kokkos array. I don't think there is anywhere in the code that that won't work (although maybe there will be somewhere the needs to be changed from `std::array` to `auto`).

Reply:
There were a handful of places across a few files that needed updating. I only changed the ones the compiler needed, but there is some mixed usage now in those files/classes that are now using both `std::array` & `Kokkos::Array`. Not sure if people have strong feelings about that, but I was worried it might start to creep to a lot more code if I started pulling on that thread more.

Reply:
Maybe the solution (for another PR) is to define `template <class T, std::size_t N> using array = Kokkos::array<T, N>` in `kokkos_types.hpp` and search and replace `std::array` with `array` everywhere in the code (since I believe the kokkos array should be a drop in replacement).

Reply:
I think it's missing the iterator stuff that is in `std::array`

Reply:
good point, although I don't think we are using `std::array` iterators anywhere (could be wrong though).